### PR TITLE
feat: Follow multiple users endpoints

### PR DESCRIPTION
**Pull request by jpmcb**
## Description

Adds a `v1/{username}/follows` endpoint for following bulk users.

### Example: Follow multiple users

Query where neither user are already followed:
```sh
curl -X 'PUT' \
  'http://localhost:3003/v1/users/{username}/follows' \
  -H 'accept: */*' \
  -H 'Authorization: Bearer <redacted>' \
  -H 'Content-Type: application/json' \
  -d '{
  "usernames": [
    "bdougie",
    "brandonroberts"
  ]
}'
```

Result:

```json
{
  "follows": [
    {
      "id": 64,
      "user_id": 23109390,
      "following_user_id": 5713670,
      "created_at": "2023-07-13T04:03:30.544Z",
      "updated_at": "2023-10-24T23:41:40.743Z",
      "deleted_at": "2023-10-24T23:41:40.743Z"
    },
    {
      "id": 121,
      "user_id": 23109390,
      "following_user_id": 42211,
      "created_at": "2023-08-24T02:40:59.365Z",
      "updated_at": "2023-10-24T23:41:33.896Z",
      "deleted_at": "2023-10-24T23:41:33.896Z"
    }
  ],
  "errors": []
}
```

### Example: One user is already followed, the other isn't

```sh
curl -X 'PUT' \
  'http://localhost:3003/v1/users/{username}/follows' \
  -H 'accept: */*' \
  -H 'Authorization: Bearer <redacted>' \
  -H 'Content-Type: application/json' \
  -d '{
  "usernames": [
    "bdougie",
    "brandonroberts"
  ]
}'
```

Result: one error is returned for user that is already followed

```json
{
  "follows": [
    {
      "id": 64,
      "user_id": 23109390,
      "following_user_id": 5713670,
      "created_at": "2023-07-13T04:03:30.544Z",
      "updated_at": "2023-10-24T23:43:22.477Z",
      "deleted_at": "2023-10-24T23:43:22.477Z"
    }
  ],
  "errors": [
    {
      "response": {
        "statusCode": 409,
        "message": "You have already followed this user: 42211",
        "error": "Conflict"
      },
      "status": 409,
      "options": {},
      "message": "You have already followed this user: 42211",
      "name": "ConflictException"
    }
  ]
}
```

### Example: unknown user is found

```sh
curl -X 'PUT' \
  'http://localhost:3003/v1/users/{username}/follows' \
  -H 'accept: */*' \
  -H 'Authorization: Bearer <redacted>' \
  -H 'Content-Type: application/json' \
  -d '{
  "usernames": [
    "bdougie",
    "user-does-not-exist-foo-bar-baz",
    "woofwoofwoofwoofwoof"
  ]
}'
```

Result: neither of the non-existent users are found and return an error

```json
{
  "statusCode": 404,
  "message": [
    "user-does-not-exist-foo-bar-baz",
    "woofwoofwoofwoofwoof"
  ],
  "error": "Not Found"
}
```

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes https://github.com/open-sauced/api/issues/389

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: src/user/user-follow.controller.ts [hunk 1]
```diff
@@ -54,6 +56,42 @@ export class UserFollowsController {
...
+  @ApiBearerAuth()
+  @UseGuards(SupabaseGuard)
+  @ApiOperation({
+    operationId: "followUsersById",
+    summary: "Follows a number of users by username",
+  })
+  @ApiNotFoundResponse({ description: "Users not found" })
+  @ApiConflictResponse({ description: "You have already followed this user" })
+  @ApiBody({ type: FollowManyUsersDto })
+  async followUsersByUsername(
```
Comment (brandonroberts) [inline src/user/user-follow.controller.ts:69]: These should match

```suggestion
    operationId: "followUsersByUsername",
    summary: "Follows a number of users by username",
  })
  @ApiNotFoundResponse({ description: "Users not found" })
  @ApiConflictResponse({ description: "You have already followed this user" })
  @ApiBody({ type: FollowManyUsersDto })
  async followUsersByUsername(
```

File: src/user/services/user.service.ts [hunk 2]
```diff
@@ -187,6 +187,84 @@ export class UserService {
     return item;
   }
 
+  async findManyByUsernames(usernames: string[]): Promise<DbUser[]> {
+    const queryBuilder = this.baseQueryBuilder();
+    const lowerCaseUsernames = usernames.map((username) => username.toLowerCase());
+
+    queryBuilder
+      .addSelect(
```
Comment (brandonroberts) [inline src/user/services/user.service.ts:195]: Question: Do we need these extra queries for this search?

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: src/user/services/user.service.ts (hunk 2 above)
Comment (jpmcb) [reply src/user/services/user.service.ts:195]: Hmm probably not: i had just copied what was in `findOneByUsername` - can get rid of those 👍🏼

**Review by brandonroberts — APPROVED** [review]
🚢

**Pull request merged**